### PR TITLE
Task Controller and README updates

### PR DESCRIPTION
**Pull request by jasonjewik**
1. Added .dockerignore so that `make build` wouldn't go through the node modules and take forever on a Windows machine.
2. Added GET and POST requests for tasks to README.md
3. Changed task controller so that API responses contain both "data" and "error".

**POST** /api/tasks/

image

**POST** /api/tasks/ (bad request - missing required 'name' field)

image

**GET** /api/tasks/{id}

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,20 +22,57 @@ Use the `access_token` from Chrome to do requests
...
+{
+	"data": {
+      "id": <TWAIN_USER_ID>,
+      "first_name": "<user_first_name>",
+      "last_name": "<user_last_name>",
+      "picture_url": "<profile_picture_url_from_google",
+      "created_at": <CREATED_AT_TIMESTAMP>
+	},
+	"error": "<ERROR_MESSAGE>",
+}
```
Comment (alexlongerbeam) [inline README.md]: Why these changes? The Readme formatting looks kinda weird for this now

File: README.md [hunk 2]
```diff
@@ -22,20 +22,57 @@ Use the `access_token` from Chrome to do requests
...
+returns
+```
+{
+	"data": {
+		"id": <task_id>
+	}
+    "error": "<ERROR_MESSAGE>"
+}
+```
+
```
Comment (alexlongerbeam) [inline README.md]: Can we actually just return the whole task here? So this is a README and API change. (Also fix formatting here)

File: src/controllers/task.js [hunk 3]
```diff
@@ -33,49 +12,31 @@ const TaskController = (taskModel) => {
...
         const body = req.body;
+        if (!("name" in body))
+            return res.status(400).json({
+                error: "Malformed Request"
+            })
+
         const name = body.name;
         const description = body.description;
         const duration = body.duration;
         const [data, err] = await taskModel.createTask(name, description, duration);
```
Comment (alexlongerbeam) [inline src/controllers/task.js]: Instead, just check if "name" or "duration" is undefined after getting it from the body (duration should definitely be a required field)

File: src/controllers/task.js [hunk 4]
```diff
@@ -33,49 +12,31 @@ const TaskController = (taskModel) => {
...
-        Error - 400 Status Code
-        {
-            "message": string
-        }
-    */
     router.post('/', async (req, res) => {
         if (!req.body)
             return res.status(400).json({
-                message: "Malformed Request"
+                error: "Malformed Request"
```
Comment (alexlongerbeam) [inline src/controllers/task.js]: For this return as well as the error return below, can you also add an empty "data" field? Makes things more consistent if every API return has both data and error, regardless of what it is

File: README.md (hunk 1 above)
Comment (alexlongerbeam) [reply README.md]: Maybe remove one tab indent from all the data fields in the original version but I think you removed both

Comment (jasonjewik) [reply README.md]: Oh, okay, I see what you mean. For some reason it rendered differently on my computer when I was working on it earlier, but I'll definitely clean it up.

**Comment by jasonjewik** [comment]
1. Fixed formatting issues in README.md
2. Changed all API return messages to include `data` and `error` fields.
3. Changed checks for undefined fields to happen after assignment instead of before.

**POST** /api/tasks/
image

**GET** /api/tasks/{id}
image

**POST** /api/events/
image

**GET** /api/events/{id}
image

File: README.md [hunk 5]
```diff
@@ -22,20 +22,96 @@ Use the `access_token` from Chrome to do requests
...
+### Create Event
+POST /api/events/
+```
+{
+  "name": "<event_name>"
+}
+
```
Comment (alexlongerbeam) [inline README.md]: Missing an extra ``` here, the rest of the readme got formatted backwards
